sdk/python/lib/pulumi/runtime/stack.py

```diff
@@ -67,6 +67,8 @@ async def wait_for_rpcs(await_all_outstanding_tasks=True) -> None:
         # We await each RPC in turn so that this loop will actually block rather than busy-wait.
         while len(RPC_MANAGER.rpcs) > 0:
             await asyncio.sleep(0)
+            if RPC_MANAGER.unhandled_exception is not None:
```

Comment:
So this terminates the `while len(RPC_MANAGER.rpcs) > 0` loop early in case there're exceptions and jumps to `raise` handler. This *might* make sense but naively this would only fix a hang if there were infinite entries in `RPC_MANAGER.rpcs` being added or else some of these entries never fulfilled their promises, making `await RPC_MANAGER.rpcs.pop()` take indefinite time. Is that the case, then why is it happening?